kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/clients/KafkaClient.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.systemtests.clients;
+
+import java.time.Duration;
+
+/**
+ * The interface Kafka client.
+ */
+public interface KafkaClient {
+    /**
+     * Sets the namespace where the kafka client will be deployed in kubernetes.
+     *
+     * @param namespace the namespace
+     * @return the kafka client
+     */
+    KafkaClient inNamespace(String namespace);
+
+    /**
+     * Produce messages.
+     *
+     * @param topicName the topic name
+     * @param bootstrap the bootstrap
+     * @param message the message
+     * @param numOfMessages the num of messages
+     */
+    void produceMessages(String topicName, String bootstrap, String message, int numOfMessages);
+
+    /**
+     * Consume messages.
+     *
+     * @param topicName the topic name
+     * @param bootstrap the bootstrap
+     * @param message the message
+     * @param numOfMessages the num of messages
+     * @param timeout the timeout
+     * @return the string
+     */
+    String consumeMessages(String topicName, String bootstrap, String message, int numOfMessages, Duration timeout);
```

Comment:
String return type is not good.  It means the test needs to know about the format of the output created by the consumer in order to separate the different records.  That should be hidden.  I think **eventually** you'll want something like `List<Record>` where record exposed the key/value/headers, maybe offset.  That might a hard contract to fulfil.   I'm not suggesting go that now.

However, wow about List<String> as a intermediate step? where the String's contents is guaranteed to contain the record's value, but there may be other stuff surrounding it.

Reply:
Currently we are assuring the String returned by the consumer contains the correct record's value, if the message expected is not received an exception is thrown. Most of the clients produce messages adding an index in the message so we can assure that the latest one has been received properly.

I agree that a `List<Record>` could be nicer and could be a work to be done for the next release to improve the testing and I'll create the issue for that, but not for this PR.

Reply:
Ok, happy for it to be deferred.